sway/input/cursor.c

```diff
@@ -450,12 +466,12 @@ static void handle_touch_motion(struct wl_listener *listener, void *data) {
 		}
 	}
 
-	if (!surface) {
-		return;
-	}
-
-	// TODO: fall back to cursor simulation if client has not bound to touch
-	if (seat_is_input_allowed(cursor->seat, surface)) {
```

Comment:
I don't *think* that's what `seat_is_input_allowed` means. I'm pretty sure it'll return true regardless of if touch is bound, so long as the client is accepting input (of any kind?) That's my impression from playing with the tablet code, at least.

Reply:
Hmm yeah, I was just assuming from the comment that this is what it does - and for my main use case (using touch to switch windows using their titlebars) it works. As basically all apps I use, use gtk they all bind to touch anyway so it's hard to test. But I'll have a closer look at that.

Reply:
Looking at the code, I definitely think you're right - now I "only" have to find the right way to check for that, if anyone has an idea, it would be nice to know :)

Reply:
`seat_is_input_allowed` is used for lockscreens: when a lockscreen is active input events can only be sent to the lockscreen client.

Reply:
At least Wayland-native don't bind touch, e.g. kitty or alacritty, but I don't know what the check would look like. Tablets support has `wlr_surface_accepts_tablet_v2`, but I couldn't find an touch analog in a quick grep.

Reply:
Yeah, i think we need to add `wlr_surface_accepts_touch` in wlroots.